org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java

```diff
@@ -155,6 +158,39 @@ public void apply(final EObject element, IModificationContext context) throws Ba
 		createLinkingIssueResolutions(issue, acceptor);
 	}
 
+	@Fix(XtextLinkingDiagnosticMessageProvider.UNRESOLVED_TERMINAL_RULE)
+	public void fixUnresolvedTerminalRule(final Issue issue, IssueResolutionAcceptor acceptor) {
+		final String ruleName = issue.getData()[0];
```

Comment:
Could you please extract the common parts into  a single method? fixUnresolvedTerminalRule and fixUnresolvedTerminalFragmentRule are almost the same.